### PR TITLE
Potentially fix execing things in main

### LINKED ISSUE #386: Fix the `exec`
TODO: more info

(created by lyxal [here](https://chat.stackexchange.com/transcript/message/59863568))

### PR DESCRIPTION
**Pull request by Lyxal**
Closes #386

### CODE REVIEW

**Review by PyGamer0 — APPROVED** [review]

**Review by Tams-Tams — APPROVED** [review]
I dunno, looks ok. This counts as 1/2 approved.

**Review by AMiller42 — APPROVED** [review]
imo, lgtm, but idk tbh.

File: setup.py [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 setup(
     name="vyxal",
-    version="2.6.1",
+    version="2.7.0",
```
Comment (pxeger) [inline setup.py:5]: a version bump should be separate from any other changes

File: vyxal/main.py [hunk 2]
```diff
@@ -143,7 +143,8 @@ def execute_vyxal(file_name, flags, inputs, output_var=None, online_mode=False):
 
     ctx.stacks.append(stack)
     try:
-        exec(code)
+        locals()["stack"] = stack
```
Comment (pxeger) [inline vyxal/main.py:146]: assigning to `locals()[]` is technically undefined behaviour in Python

File: vyxal/main.py [hunk 3]
```diff
@@ -143,7 +143,8 @@ def execute_vyxal(file_name, flags, inputs, output_var=None, online_mode=False):
...
+        exec(code, locals() | globals())
```
Comment (pxeger) [inline vyxal/main.py:147]: Merging these creates a new object, so any changes to the new object don't affect the old `locals()` or `globals()`. (i.e., any assignments inside `code` won't affect any variables in `execute_vyxal()`). Is this what you want?

If not, use a `collections.chainmap`.

**Review by pxeger — CHANGES_REQUESTED** [review]

File: setup.py (hunk 1 above)
Comment (Lyxal) [reply setup.py:5]: I figured that too, but I realised I forgot to do it before actually making the release notes so I quickly placed it here.

File: vyxal/main.py (hunk 2 above)
Comment (Lyxal) [reply vyxal/main.py:146]: Looks like it isn't needed anyway. Good catch.

File: vyxal/main.py (hunk 3 above)
Comment (Lyxal) [reply vyxal/main.py:147]: I'm fine with it not actually changing locals or global. In fact, that's probably better in the bigger picture.

**Review by cgccuser — APPROVED** [review]

**Review by pxeger — APPROVED** [review]

**Pull request merged**